mne/beamformer/tests/test_lcmv.py

```diff
@@ -666,46 +670,111 @@ def test_localization_bias_fixed(bias_params_fixed, reg, weight_norm, use_cov,
-        (0.00, 'max-power', 'nai', True, None, 43, 50),
+        (0.00, 'vector', None, True, None, 23, 24, 0.96, 0.97),
+        (0.00, 'vector', 'unit-noise-gain-invariant', True, None, 52, 54, 0.95, 0.96),  # noqa: E501
+        (0.00, 'vector', 'unit-noise-gain', True, None, 44, 45, 0.97, 0.98),
+        (0.00, 'vector', 'nai', True, None, 44, 45, 0.97, 0.98),
+        (0.00, 'max-power', None, True, None, 14, 15, 0, 0),
+        (0.00, 'max-power', 'unit-noise-gain-invariant', True, None, 35, 37, 0, 0),  # noqa: E501
+        (0.00, 'max-power', 'unit-noise-gain', True, None, 35, 37, 0, 0),
+        (0.00, 'max-power', 'nai', True, None, 35, 37, 0, 0),
     ])
 def test_localization_bias_free(bias_params_free, reg, pick_ori, weight_norm,
-                                use_cov, depth, lower, upper):
+                                use_cov, depth, lower, upper,
+                                lower_ori, upper_ori):
     """Test localization bias for free-orientation LCMV."""
     evoked, fwd, noise_cov, data_cov, want = bias_params_free
     if not use_cov:
         evoked.pick_types(meg='grad')
         noise_cov = None
-    loc = apply_lcmv(evoked, make_lcmv(evoked.info, fwd, data_cov, reg,
-                                       noise_cov, pick_ori=pick_ori,
-                                       weight_norm=weight_norm,
-                                       depth=depth)).data
+    with pytest.warns(None):  # rank deficiency of data_cov
+        filters = make_lcmv(evoked.info, fwd, data_cov, reg,
+                            noise_cov, pick_ori=pick_ori,
+                            weight_norm=weight_norm,
+                            depth=depth)
+    loc = apply_lcmv(evoked, filters).data
+    if pick_ori == 'vector':
+        ori = loc.copy() / np.linalg.norm(loc, axis=1, keepdims=True)
+    else:
+        # doesn't make sense for pooled (None) or max-power (can't be all 3)
+        ori = None
     loc = np.linalg.norm(loc, axis=1) if pick_ori == 'vector' else np.abs(loc)
     # Compute the percentage of sources for which there is no loc bias:
-    perc = (want == np.argmax(loc, axis=0)).mean() * 100
+    max_idx = np.argmax(loc, axis=0)
+    perc = (want == max_idx).mean() * 100
+    assert lower <= perc <= upper
+    _assert_free_ori_match(ori, max_idx, lower_ori, upper_ori)
+
+
+# Changes here should be synced with the ones above, but these have meaningful
+# orientation values
+@pytest.mark.parametrize(
+    'reg, weight_norm, use_cov, depth, lower, upper, lower_ori, upper_ori', [
+        (0.05, 'unit-noise-gain-invariant', False, None, 38, 40, 0.52, 0.54),
+        (0.05, 'unit-noise-gain', False, None, 38, 40, 0.52, 0.54),
+        (0.05, 'nai', True, None, 56, 57, 0.56, 0.58),
+        (0.05, None, True, None, 27, 28, 0.54, 0.56),
+        (0.05, None, True, 0.8, 42, 43, 0.54, 0.56),
+        # no reg
+        (0.00, None, True, None, 50, 51, 0.57, 0.58),
```

Comment:
@britta-wstnr I implemented the test, but it doesn't do too well. The mean absolute dot product for random unit vectors converges to 0.5, a perfect match will be 1.0, and these hover between 0.53-0.57. Maybe it's because there are too many sources active? Though the non-ori-selection / full vector solutions don't seem to have this problem.

It's also possible that the covariance matrix I'm passing is not the best, even though it's the product of the simulated (normal-orientation) source lead fields with itself transposed. But this will at least help us tell if we break something with our orientation selection.

Reply:
Mh I am puzzled about this, but I am also not sure I fully understood how you simulated the data. But fine to just let it in passing for now, and we can investigate later.